### PR TITLE
EVG-18311: add more logging to json parse failure

### PR DESCRIPTION
**Pull request by dylrich**
Right now failure to parse the response body from a login request gives an error message without much useful context because the body isn't included. This commit fixes that by logging the body as well.

### CODE REVIEW

**Review by Kimchelly — CHANGES_REQUESTED** [review]
Is there a Jira ticket associated with this change?

Can we also include a PR description summarizing what this change is hoping to accomplish?

**Comment by dylrich** [comment]
Ah, sorry, didn't realize we needed tickets for this project. I'll file one and fix.

**Comment by dylrich** [comment]
Which project is this associated with, do you know?

**Comment by dylrich** [comment]
OK, I filed an EVG ticket for this and think I've fixed everything up for you. I hope that's OK!

**Comment by dylrich** [comment]
Sounds good to me!

File: barquesubmit/client.go
Comment (Kimchelly) [inline barquesubmit/client.go:114]: Part of the contract of using the response struct is that [the docs say we have to close the response body](https://pkg.go.dev/net/http#Response). Can we put a `defer resp.Body.Close()` on L108?

**Review by Kimchelly — APPROVED** [review]
LGTM mod one comment about closing the response body.

**Comment by dylrich** [comment]
We ended up figuring out our problem without having the full error message - sorry for the busywork. I'll make that change anyway!

**Comment by Kimchelly** [comment]
The code seems to be the same as before the push? I'll merge once it has the one defer line.

**Comment by dylrich** [comment]
Sorry about that! Got distracted

**Pull request merged**
